### PR TITLE
CNF-14786:  Validate switching to a new hardware profile

### PR DESCRIPTION
**Pull request by irinamihai**
Description:
- provide readme steps for going through the update of the hardware profile used by a ProvisioningRequest

### CODE REVIEW

File: docs/cluster-configuration.md [hunk 1]
```diff
@@ -427,3 +427,94 @@ metadata:
...
+5. The hardware plugin calls the hardware manager passing the new hardware profile from the `Node` `spec`.
+6. The hardware manager updates the profile.
+7. The hardware plugin polls the status of the returned JobId from the hardware manager.
+    * Success scenario:
+        * The hardware plugin updates the status of the `Node` CR.
+    * Failure scenario:
+        * The operation is aborted.
+        * The `Node` CR is updated with the failure reason which is propagated up to the `ProvisioningRequest`.
+        * The O-Cloud manager does not initiate a rollback of any nodes already updates. This is left to the user to remediate.
+8. Once all nodes have been updated, the hardware plugin will update the status of the `NodePool` CR `Configured` condition is updated to reflect the result of the operation:
```
Comment (donpenney) [inline docs/cluster-configuration.md:499]: This flow sounds correct to me

File: docs/cluster-configuration.md [hunk 2]
```diff
@@ -427,3 +427,94 @@ metadata:
...
+        status: "False"
+        type: Configured
+    ```
+    * Obtains the list of nodes from the `NodePool` CR for the master MCP.
+    * For the SNO case that we are considering, there is only one node that cannot be cordoned and drained.
+    * Updates `spec.hwProfile` in the `Node` (`node.o2ims-hardwaremanagement.oran.openshift.io/v1alpha1`) CR.
+
+5. The hardware plugin calls the hardware manager passing the new hardware profile from the `Node` `spec`.
+6. The hardware manager updates the profile.
+7. The hardware plugin polls the status of the returned JobId from the hardware manager.
```
Comment (bartwensley) [inline docs/cluster-configuration.md:492]: Should this be more generic? Something like:
The hardware plugin waits for the result from the hardware manager.

File: docs/cluster-configuration.md [hunk 3]
```diff
@@ -427,3 +427,94 @@ metadata:
...
+        message: Spec updated; awaiting configuration application by the hardware plugin
+        reason: ConfigurationUpdateRequested
+        status: "False"
+        type: Configured
+    ```
+    * Obtains the list of nodes from the `NodePool` CR for the master MCP.
+    * For the SNO case that we are considering, there is only one node that cannot be cordoned and drained.
+    * Updates `spec.hwProfile` in the `Node` (`node.o2ims-hardwaremanagement.oran.openshift.io/v1alpha1`) CR.
+
+5. The hardware plugin calls the hardware manager passing the new hardware profile from the `Node` `spec`.
```
Comment (bartwensley) [inline docs/cluster-configuration.md:490]: Maybe more specific would be good - something like:
The hardware plugin requests the hardware manager to apply the new hardware profile from the `Node` `spec`.

File: docs/cluster-configuration.md [hunk 4]
```diff
@@ -427,3 +427,94 @@ metadata:
...
+6. The hardware manager updates the profile.
+7. The hardware plugin polls the status of the returned JobId from the hardware manager.
+    * Success scenario:
+        * The hardware plugin updates the status of the `Node` CR.
+    * Failure scenario:
+        * The operation is aborted.
+        * The `Node` CR is updated with the failure reason which is propagated up to the `ProvisioningRequest`.
+        * The O-Cloud manager does not initiate a rollback of any nodes already updates. This is left to the user to remediate.
+8. Once all nodes have been updated, the hardware plugin will update the status of the `NodePool` CR `Configured` condition is updated to reflect the result of the operation:
```
Comment (bartwensley) [inline docs/cluster-configuration.md:499]: Wording: remove "is updated"

File: docs/cluster-configuration.md [hunk 5]
```diff
@@ -427,3 +427,94 @@ metadata:
...
+    * Updates `spec.hwProfile` in the `Node` (`node.o2ims-hardwaremanagement.oran.openshift.io/v1alpha1`) CR.
+
+5. The hardware plugin calls the hardware manager passing the new hardware profile from the `Node` `spec`.
+6. The hardware manager updates the profile.
+7. The hardware plugin polls the status of the returned JobId from the hardware manager.
+    * Success scenario:
+        * The hardware plugin updates the status of the `Node` CR.
+    * Failure scenario:
+        * The operation is aborted.
+        * The `Node` CR is updated with the failure reason which is propagated up to the `ProvisioningRequest`.
```
Comment (tliu2021) [inline docs/cluster-configuration.md:497]: Should this be "The NodePool CR is updated with the failure reason"? The O-Cloud manager only propagated the NodePool status to the 'ProvisioningRequest'.

**Pull request merged**